### PR TITLE
Added _ips

### PR DESCRIPTION
**Pull request by QuentinPerez**
#178 

Display the IPs
```bash
./scw _ips | anonuuid
{
  "ips": [
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "11111111-1111-1111-1111-111111111111",
      "server": {
        "id": "22222222-2222-1222-2222-222222222222",
        "name": "buildwordpress"
      },
      "address": "212.47.239.7"
    }
  ]
}
```

Display an IP
```bash
./scw _ips 11111111-1111-1111-1111-111111111111 | anonuuid
{
  "ip": {
    "organization": "00000000-0000-1000-0000-000000000000",
    "reverse": "",
    "id": "11111111-1111-1111-1111-111111111111",
    "server": {
      "id": "22222222-2222-1222-2222-222222222222",
      "name": "buildwordpress"
    },
    "address": "212.47.239.7"
  }
}
```

New IP
```bash
./scw _ips -n
{
  "ip": {
    "organization": "00000000-0000-1000-0000-000000000000",
    "reverse": "",
    "id": "11111111-1111-1111-1111-111111111111",
    "server": {},
    "address": "212.47.249.205"
  }
}

./scw _ips | anonuuid
{
  "ips": [
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "11111111-1111-1111-1111-111111111111",
      "server": {},
      "address": "212.47.249.205"
    },
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "22222222-2222-1222-2222-222222222222",
      "server": {
        "id": "33333333-3333-1333-3333-333333333333",
        "name": "buildwordpress"
      },
      "address": "212.47.239.7"
    }
  ]
}
```

Delete an IP
```bash
./scw _ips -d 09591c34-8b87-4a99-a31b-d073020e1007
./scw _ips | anonuuid
{
  "ips": [
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "11111111-1111-1111-1111-111111111111",
      "server": {
        "id": "22222222-2222-1222-2222-222222222222",
        "name": "buildwordpress"
      },
      "address": "212.47.239.7"
    }
  ]
}
```

Attach an IP
```bash
./scw _ips -a 11111111-1111-1111-1111-111111111111 22222222-2222-1222-2222-222222222222
./scw _ips | anonuuid
{
  "ips": [
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "11111111-1111-1111-1111-111111111111",
      "server": {
        "id": "22222222-2222-1222-2222-222222222222",
        "name": "scw-e34a94"
      },
      "address": "212.47.249.205"
    },
    {
      "organization": "00000000-0000-1000-0000-000000000000",
      "reverse": "",
      "id": "33333333-3333-1333-3333-333333333333",
      "server": {
        "id": "44444444-4444-1444-4444-444444444444",
        "name": "buildwordpress"
      },
      "address": "212.47.239.7"
    }
  ]
}
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1131,12 +1131,17 @@ $ scw inspect myserver | jq '.[0].public_ip.address'
 
 ### master (unreleased)
 
-* Report **permissions** in `scw info` ([#191](https://github.com/scaleway/scaleway-cli/issues/191))
+* Added _ips
```
Comment (moul) [inline README.md]: Rename `* Support of 'scw _ips' command ...`

File: README.md [hunk 2]
```diff
@@ -1131,12 +1131,17 @@ $ scw inspect myserver | jq '.[0].public_ip.address'
...
+([#196](https://github.com/scaleway/scaleway-cli/issues/196))
```
Comment (moul) [inline README.md]: Do not break those lines, a changelog may be copied without markdown support

**Comment by moul** [comment]
LGTM for "get ips", now we need to implement some actions like attach ip to a server which is what is blocking @arianvp

**Comment by moul** [comment]
About the "attach ip to server", you can do it from the `scw _patch server:...` if you want, we can keep `scw _ips` just for listing if you prefer

File: pkg/cli/x_ips.go [hunk 3]
```diff
@@ -0,0 +1,152 @@
...
+	UsageLine: "_ips [OPTIONS] [ARGS]",
+
+	Description: "",
+	Hidden:      true,
+	Help:        "Display your ips",
+	Examples: `
+    $ scw _ips
+    $ scw _ips IP_ID
+    $ scw _ips --new
+    $ scw _ips --update IP_ID:SERVER_ID
```
Comment (moul) [inline pkg/cli/x_ips.go]: I think, two parameters would be better than 1 parameter with a `colon` separator

@aimxhaisse what do you think ?

File: pkg/cli/x_ips.go [hunk 4]
```diff
@@ -0,0 +1,152 @@
...
+	"github.com/scaleway/scaleway-cli/pkg/api"
+)
+
+var cmdIPS = &Command{
+	Exec:      runIPS,
+	UsageLine: "_ips [OPTIONS] [ARGS]",
+
+	Description: "",
+	Hidden:      true,
+	Help:        "Display your ips",
```
Comment (moul) [inline pkg/cli/x_ips.go]: Update the help message

File: pkg/cli/x_ips.go [hunk 5]
```diff
@@ -0,0 +1,152 @@
...
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/scaleway/scaleway-cli/pkg/api"
+)
+
+var cmdIPS = &Command{
+	Exec:      runIPS,
+	UsageLine: "_ips [OPTIONS] [ARGS]",
```
Comment (moul) [inline pkg/cli/x_ips.go]: Update the usage line

File: pkg/cli/x_ips.go [hunk 6]
```diff
@@ -0,0 +1,152 @@
...
+			ID           string  `json:"id"`
+			Reverse      *string `json:"reverse"`
+			Organization string  `json:"organization"`
+			Server       string  `json:"server"`
+		}
+		update.Address = ip.IP.Address
+		update.ID = ip.IP.ID
+		update.Organization = ip.IP.Organization
+		update.Server = newParts[1]
+		resp, err := cmd.API.PutResponse(fmt.Sprintf("ips/%s", newParts[0]), update)
```
Comment (moul) [inline pkg/cli/x_ips.go]: Use an helper in the pkg/api

File: pkg/cli/x_ips.go [hunk 7]
```diff
@@ -0,0 +1,152 @@
...
+		err = decoder.Decode(&error)
+		if err != nil {
+			return err
+		}
+		error.StatusCode = resp.StatusCode
+		error.Debug()
+		return error
+	}
+	if ipDelete != "" {
+		resp, err := cmd.API.DeleteResponse(fmt.Sprintf("ips/%s", ipDelete))
```
Comment (moul) [inline pkg/cli/x_ips.go]: Use an helper in **pkg/api**

File: pkg/cli/x_ips.go [hunk 8]
```diff
@@ -0,0 +1,152 @@
...
+func runIPS(cmd *Command, args []string) error {
+	if ipHelp {
+		return cmd.PrintUsage()
+	}
+	if ipNew {
+		var orga struct {
+			Organization string `json:"organization"`
+		}
+		orga.Organization = cmd.API.Organization
+		resp, err := cmd.API.PostResponse("ips", orga)
```
Comment (moul) [inline pkg/cli/x_ips.go]: Use an helper in **pkg/api**

File: pkg/cli/x_ips.go (hunk 3 above)
Comment (QuentinPerez) [inline pkg/cli/x_ips.go]: I used `colon` here https://github.com/scaleway/scaleway-cli/blob/master/pkg/cli/x_security-groups.go#L26-L33

Comment (moul) [inline pkg/cli/x_ips.go]: Yes, that's why I'm not 100% sure it's fully wrong

I think colon is useful in two cases:

* mapping, as docker do for volumes: `docker run -v /local/path:/container/path image`
* complex identifier but still only one object, i.e: "server:123456"

---

In this particular case, I think --update IP_ID:SERVER_ID is not clear at all and I would move the function to `scw _patch IP_ID server=SERVER_ID` or `scw _ips --attach IP_ID SERVER_ID` without colon

About the security group, I think the case is too complex for now, I suggest to wait a little and refactor later

**Comment by moul** [comment]
As the method is prefixed by `_`, I'm okay to merge it soon and open a new ticket to improve globally `_ips` support

But you need to move the API calls in the `pkg/api`, this is mandatory as it is exactly what @arianvp is looking for

File: pkg/api/api.go [hunk 9]
```diff
@@ -1872,6 +1891,133 @@ func (s *ScalewayAPI) GetASecurityGroup(groupsID string) (*ScalewayGetSecurityGr
...
+	if err != nil {
+		return nil, err
+	}
+	error.StatusCode = resp.StatusCode
+	error.Debug()
+	return nil, error
+}
+
+// AttachIP attachs an IP to a server
+func (s *ScalewayAPI) AttachIP(idIP, serverID string) error {
```
Comment (moul) [inline pkg/api/api.go]: Be uniform, use always **ipID**

**Comment by moul** [comment]
LGTM

File: pkg/api/api.go (hunk 9 above)
Comment (QuentinPerez) [inline pkg/api/api.go]: Oups mistake `idIP` -> `ipID` 😄

**Pull request merged**